.github/workflows/ci.yml

```diff
@@ -49,10 +60,16 @@ jobs:
           ini-values: zend.assertions=1, error_reporting=-1, display_errors=On
           coverage: xdebug
 
+      - name: 'Composer: update PHPUnit for testing lowest'
+        if: ${{ matrix.dependency-versions == 'lowest' }}
+        run: composer require --no-update phpunit/phpunit:"^9.1"
```

Comment:
changing the requirement before running with lowest deps defeats the goal of this job, which is to ensure that the lowest bound of the requirements is the right one.

We might instead add a conflict rule for `phpunit/phpunit <9.1`, which still keeps the requirement untouched.

Reply:
Wouldn't a conflict rule prevent installation in combination with PHPUnit < 9.1 ?

Reply:
a conflict rule being added here for this job (instead of replacing the `require` rule as you do), not a rule added in the committed composer.json

Reply:
Ah! In that case, my question changes: how can I set a `conflict` rule from the command-line ? I'm not aware of any Composer CLI command available to do this. Interested to learn this new trick ;-)

Reply:
I don't think the CLI allows to do it. You will need to do some sed magic on the composer.json file.

Reply:
Hmm.. that sounds like a very complicated way to achieve the same thing as is being achieved via the above command...

I suppose I could do it by adding a second `test-lowest-composer.json` file to the repo and conditionally using that during the CI runs, but that would risk that file getting out-of-sync with the _real_ `composer.json` files.

Other than that, I'm out of ideas. I don't have the set up to create or test `sed` commands, so getting that working will be _challenging_.

Reply:
The current setup does not catch mistakes in the lowest bound in the composer.json, which is the goal of a `--prefer-lowest` CI job. Using a separate composer.json would not help either unless you have a test enforcing it to stay in sync with the normal composer.json.
As we don't have a `--prefer-lowest` job in our current CI setup, another solution is to avoid adding such a job in your PR (as that's not its main goal)

Reply:
> The current setup does not catch mistakes in the lowest bound in the composer.json, which is the goal of a --prefer-lowest CI job. 

Actually, no. My _intention_ with adding the `lowest` builds was to ensure that the _real_ trait works as advertised in the high/low supported versions of PHPUnit. Nothing more.

But yeah, I can remove it from this PR for now and you can revisit this later.